java/ql/src/Security/CWE/CWE-312/AllowBackupAttributeEnabled.ql

```diff
@@ -0,0 +1,28 @@
+/**
+ * @name Android allowBackup attribute enabled
+ * @description Android manifests which do not disable the `android:allowBackup` attribute allow backups, which can store sensitive information.
+ * @kind problem
+ * @problem.severity recommendation
+ * @security-severity 7.5
+ * @id java/android/allowbackup-enabled
+ * @tags security
+ *       external/cwe/cwe-312
+ * @precision very-high
+ */
+
+import java
+import semmle.code.xml.AndroidManifest
+
+from AndroidApplicationXmlElement androidAppElem
+where
+  not androidAppElem.getFile().(AndroidManifestXmlFile).isInBuildDirectory() and
```

Comment:
I'm thinking that we could move all this logic into `allowsBackup`. That offers several advantages:

* We hide complexity from the user.
* The predicate can be transparently reused in the test.
* We can make private at least `allowsBackupExplicitly`, and maybe `providesMainIntent`, since they are only used by `allowsBackup`.

If you apply this suggestion, you'd also need to adjust the change note under `lib/`.

What do you think?

Reply:
Ok that makes sense. Especially privating `allowsBackupExplicitly`